### PR TITLE
Add single-replica metadata support

### PR DESCRIPTION
**Pull request by kjsanger**

### CODE REVIEW

**Comment by dkj** [comment]
Should the test be inheriting from `WTSI::NPG::HTS::Test` to ensure environment variable `WTSI_NPG_iRODS_Test_IRODS_ENVIRONMENT_FILE` is set and used to determine the test iRODS service?

**Comment by dkj** [comment]
Should the test be inheriting from `WTSI::NPG::HTS::Test` to ensure environment variable `WTSI_NPG_iRODS_Test_IRODS_ENVIRONMENT_FILE` is set and used to determine the test iRODS service?

**Comment by kjsanger** [comment]
> Should the test be inheriting from `WTSI::NPG::HTS::Test` to ensure environment variable `WTSI_NPG_iRODS_Test_IRODS_ENVIRONMENT_FILE` is set and used to determine the test iRODS service?

It does https://github.com/wtsi-npg/npg_irods/pull/359/files#diff-bd94e5bf43f7d8603c6a06e44aa5f7caa265aa1276c7f86b927c00f42386bcbdR13

**Comment by dkj** [comment]
> It does https://github.com/wtsi-npg/npg_irods/pull/359/files#diff-bd94e5bf43f7d8603c6a06e44aa5f7caa265aa1276c7f86b927c00f42386bcbdR13

Thanks!

**Comment by dkj** [comment]
> It does https://github.com/wtsi-npg/npg_irods/pull/359/files#diff-bd94e5bf43f7d8603c6a06e44aa5f7caa265aa1276c7f86b927c00f42386bcbdR13

Thanks!

File: t/lib/WTSI/NPG/Data/SingleReplicaMetaUpdaterTest.pm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  my $obj2 =
+    WTSI::NPG::iRODS::DataObject->new($irods,
+                                      "$irods_tmp_coll/single_replica/2.txt");
+  $obj2->supersede_avus($ebi_md5, 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');
+
+  my ($num_objs, $num_processed, $num_errors) =
+    $m->update_single_replica_metadata(end_date => $middle);
+  is($num_objs, 4, 'Expected 4 objects found');
+  is($num_processed, 4, 'Expected 4 objects processed');
+  is($num_errors, 1, 'Expected no errors');
```
Comment (dkj) [inline t/lib/WTSI/NPG/Data/SingleReplicaMetaUpdaterTest.pm:137]:
```suggestion
  is($num_errors, 1, 'Expected 1 errors');
```

**Review by dkj — DISMISSED** [review]
Tests cover scenarios. Text for one test best updated.

**Pull request merged**